automat/_visualize.py

```diff
@@ -1,30 +1,70 @@
+import graphviz
 
-from __future__ import unicode_literals
 
 def _gvquote(s):
     return '"{}"'.format(s.replace('"', r'\"'))
 
+
 def _gvhtml(s):
     return '<{}>'.format(s)
 
 
-def graphviz(automaton, inputAsString=repr,
-             outputAsString=repr,
-             stateAsString=repr):
+def elementMaker(name, *children, **attrs):
+    """
+    Construct a string from the HTML element description.
     """
-    Produce a graphviz dot file as an iterable of bytes.
+    formattedAttrs = ' '.join('{}={}'.format(key, _gvquote(str(value)))
+                              for key, value in sorted(attrs.items()))
+    formattedChildren = ''.join(children)
+    return u'<{name} {attrs}>{children}</{name}>'.format(
+        name=name,
+        attrs=formattedAttrs,
+        children=formattedChildren)
 
-    Use like so::
 
-        with f as open('something', 'wb'):
-            f.writelines(graphviz(automaton))
+def tableMaker(inputLabel, outputLabels, port, _E=elementMaker):
+    """
+    Construct an HTML table to label a state transition.
     """
-    yield "digraph {\n"
-    # yield "pad=0.25;\n"
-    # yield 'splines="curved";\n'
-    yield "graph[pack=true,dpi=100]\n"
-    yield 'node[fontname="Menlo"]\n'
-    yield 'edge[fontname="Menlo"]\n'
+    colspan = {}
+    if outputLabels:
+        colspan['colspan'] = str(len(outputLabels))
+
+    inputLabelCell = _E("td",
+                        _E("font",
+                           inputLabel,
+                           face="menlo-italic"),
+                        color="purple",
+                        port=port,
+                        **colspan)
+
+    pointSize = {"point-size": "9"}
+    outputLabelCells = [_E("td",
+                           _E("font",
+                              outputLabel,
+                              **pointSize),
+                           color="pink")
+                        for outputLabel in outputLabels]
+
+    rows = [_E("tr", inputLabelCell)]
+
+    if outputLabels:
+        rows.append(_E("tr", *outputLabelCells))
+
+    return _E("table", *rows)
+
+
+def makeDigraph(automaton, inputAsString=repr,
+                outputAsString=repr,
+                stateAsString=repr):
+    """
+    Produce a C{graphviz.Digraph} object from an automaton.
```

Comment:
Technically speaking the right syntax here would be `L{}`.  Is the `C{}` necessary to work around pydoctor complaints?